src/pipelines/masked_language.rs

```diff
@@ -0,0 +1,581 @@
+                        input_embeds,
+                        encoder_hidden_states,
+                        encoder_mask,
+                        train,
+                    )
+                    .prediction_scores
+            }
+            Self::FNet(ref model) => {
+                model
+                    .forward_t(input_ids, token_type_ids, position_ids, input_embeds, train)
+                    .expect("Error in FNet forward pass.")
+                    .prediction_scores
+            }
+        }
+    }
+}
+
+/// # MaskedLanguageModel for Masked Language (e.g. Fill Mask)
+pub struct MaskedLanguageModel {
+    tokenizer: TokenizerOption,
+    language_encode: MaskedLanguageOption,
+    var_store: VarStore,
+    max_length: usize,
+}
+
+impl MaskedLanguageModel {
+    /// Build a new `MaskedLanguageModel`
+    ///
+    /// # Arguments
+    ///
+    /// * `config` - `MaskedLanguageConfig` object containing the resource references (model, vocabulary, configuration) and device placement (CPU/GPU)
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # fn main() -> anyhow::Result<()> {
+    /// use rust_bert::pipelines::masked_language::MaskedLanguageModel;
+    ///
+    /// let model = MaskedLanguageModel::new(Default::default())?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn new(config: MaskedLanguageConfig) -> Result<MaskedLanguageModel, RustBertError> {
+        let config_path = config.config_resource.get_local_path()?;
+        let vocab_path = config.vocab_resource.get_local_path()?;
+        let weights_path = if config.model_local_resource.is_none() {
+            config.model_resource.unwrap().get_local_path()?
+        } else {
+            config.model_local_resource.unwrap().get_local_path()?
+        };
+        let merges_path = if let Some(merges_resource) = &config.merges_resource {
+            Some(merges_resource.get_local_path()?)
+        } else {
+            None
+        };
+        let device = config.device;
+
+        let tokenizer = TokenizerOption::from_file(
+            config.model_type,
+            vocab_path.to_str().unwrap(),
+            merges_path.as_deref().map(|path| path.to_str().unwrap()),
+            config.lower_case,
+            config.strip_accents,
+            config.add_prefix_space,
+        )?;
+        let mut var_store = VarStore::new(device);
+        let model_config = ConfigOption::from_file(config.model_type, config_path);
+        let max_length = model_config
+            .get_max_len()
+            .map(|v| v as usize)
+            .unwrap_or(usize::MAX);
+
+        let language_encode =
+            MaskedLanguageOption::new(config.model_type, &var_store.root(), &model_config)?;
+        var_store.load(weights_path)?;
+        Ok(MaskedLanguageModel {
+            tokenizer,
+            language_encode,
+            var_store,
+            max_length,
+        })
+    }
+
+    fn prepare_for_model<'a, S>(&self, input: S) -> Tensor
```

Comment:
I believe this should set the attention mask and mask the `mask_token` positions as well (probably return a tuple of tensors)